articles/build-an-outlook-clone-using-react-hooks/index.md

```diff
@@ -0,0 +1,881 @@
+   ```jsx
+    const refVariable = useRef(initialValue)
+   ```
+   It is usually used to access the DOM variables and to keep track of a mutable variable.
+4. **createContext** and **useContext**: createContext us used to pass props from parent to child components. Consider the example of passing props from parent to child components present many levels below. Sending it via the props makes the code messy. Instead, createContext is used to send the props. Whenever the props are needed, we use useContext to get the value of the props.  We will discuss this below in the project.
+
+### Code
+Alrighty, we first begin with creating a React project, using create-react-app package. Open a new terminal of your choice and execute the following commands.  
+
+```terminal
+npx create-react-app outlookclonereacthooks
+cd outlookclonereacthooks
+npm start
+```
+
+The folder structure should look like this.
+
+```
+outlookclonereacthooks
+├── README.md
+├── node_modules
+├── package.json
+├── .gitignore
+├── public
+│   ├── favicon.ico
+│   ├── index.html
+│   ├── logo192.png
+│   ├── logo512.png
+│   ├── manifest.json
+│   └── robots.txt
+└── src
+    ├── App.css
+    ├── App.js
+    ├── App.test.js
+    ├── index.css
+    ├── index.js
+    ├── logo.svg
+    └── serviceWorker.js
+```
+We will be working with class-based components in this article. The next article will talk about using Hooks for the same. A study of the two articles is enough to get a working understanding of React. Many examples that we find on GitHub are in class-based components. Having the ability to use React Hooks in place of class-based components is a useful skill to have.
+
+
+Let us create a folder called components in the root folder. The updated directory looks like this. Add the components *Email.js*, *EmailBody.js*,*SearchBox.js*,*Scroll.js*, and create directories named *styledComponents* and *css*. Add the files *ButtonStyles.js*, *EmailBody.js* and *EmailStyle.js* to styledComponents directory. Add file *EmailBody.css* to the css folder.  
+
+```
+outlookclonereacthooks
+├── README.md
+├── node_modules
+├── package.json
+├── .gitignore
+├── public
+│   ├── favicon.ico
+│   ├── index.html
+│   ├── logo192.png
+│   ├── logo512.png
+│   ├── manifest.json
+│   └── robots.txt
+└── src
+    ├── components     
+    │   ├── Email.js
+    │   ├── EmailBody.js
+    │   ├── Scroll.js
+    |   ├── styledComponents
+    |      ├── ButtonStyles.js
+    │      ├── EmailBodyStyles.js
+    │      ├── EmailStyle.js
+    |   ├── css
+    |      ├── EmailBody.css
+    ├── App.css
+    ├── App.js
+    ├── App.test.js
+    ├── index.css
+    ├── index.js
+    ├── logo.svg
+    └── serviceWorker.js
+```
+
+### Setting up API on localhost
+For the project, we need to make API calls to get the list of emails and email body. Thus, lets set that up. We will be using *json-server* package. This
```

Comment:
> This

Incomplete sentence that will hopefully explain what the following command is doing.